### PR TITLE
#818 fix hasContentInAnyOrder

### PR DESCRIPTION
**Pull request by blizznets**

### CODE REVIEW

File: testing/src/org/immutables/check/IterableChecker.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public final void isOf(T... elements) {
   }
 
   public void hasContentInAnyOrder(Iterable<?> elements) {
-    verifyUsingMatcher((Matcher<? super Z>) Matchers.containsInAnyOrder(ImmutableList.copyOf(elements)));
+    verifyUsingMatcher(Matchers.containsInAnyOrder(ImmutableList.copyOf(elements).toArray()));
```
Comment (Stephan202) [inline testing/src/org/immutables/check/IterableChecker.java]: `Iterables.toArray` may avoid a second copy.

Comment (blizznets) [reply testing/src/org/immutables/check/IterableChecker.java]: @Stephan202 thnx for review, fixed

File: testing/src/org/immutables/check/IterableChecker.java [hunk 2]
```diff
@@ -70,8 +70,8 @@ public final void isOf(T... elements) {
     verifyUsingMatcher(Matchers.contains(elements));
   }
 
-  public void hasContentInAnyOrder(Iterable<?> elements) {
-    verifyUsingMatcher((Matcher<? super Z>) Matchers.containsInAnyOrder(ImmutableList.copyOf(elements)));
+  public void hasContentInAnyOrder(Iterable<T> elements) {
+    verifyUsingMatcher(Matchers.containsInAnyOrder((T[]) Iterables.toArray(elements, Object.class)));
```
Comment (elucash) [inline testing/src/org/immutables/check/IterableChecker.java]: cannot verify right now, is there any warnings to suppress `@SuppressWarnings("unchecked")` or can cast `(T[])` be removed? just asking. Thank you or PR!

Comment (blizznets) [reply testing/src/org/immutables/check/IterableChecker.java]: @elucash you're right, I'd removed redundant cast

File: testing/src/org/immutables/check/IterableChecker.java [hunk 3]
```diff
@@ -70,8 +70,8 @@ public final void isOf(T... elements) {
     verifyUsingMatcher(Matchers.contains(elements));
   }
 
-  public void hasContentInAnyOrder(Iterable<?> elements) {
-    verifyUsingMatcher((Matcher<? super Z>) Matchers.containsInAnyOrder(ImmutableList.copyOf(elements)));
+  public void hasContentInAnyOrder(Iterable<T> elements) {
```
[the remark this answers is not in the document]
Comment (blizznets) [reply testing/src/org/immutables/check/IterableChecker.java]: @Stephan202 totally agreed, and again thnx for review. if we've already touched this - it's better to make it perfect :)

**Comment by blizznets** [comment]
@elucash looks cool! I'll take a look more detailed..

**Comment by elucash** [comment]
@blizznets @Stephan202 thank you for PR and review!

**Comment by asereda-gs** [comment]
@elucash `nextgen` repository is not public. I also wanted to check your Truth / AssertJ alternative.

**Comment by elucash** [comment]
@asereda-gs invited you to the repo. I know it will be easier to just open it, just need to do some preparations etc.

**Pull request merged**